### ISSUE TITLE
Selection box is not rotated when an object is rotated

### ISSUE THREAD

**Issue by Keevle** [4ian/GDevelop#1741]
https://streamable.com/uqyxl7

This is the minor issue in action,if you drag a sprite in the editor the selection box will correctly expand,but the moment you change its angle the box dosent update properly,you can test this with any example project

**Comment by PawBud** [comment]
"I guess the "box" is a pixi object, just synchronize its angle with the object's properties". as advised by @arthuro555. I am not really acquainted with Pixi, so I would like to give this issue a try because a might learn a thing or two about Pixi JS in the process. if no one is interested, can I be guided to the relevant code?

**Comment by arthuro555** [comment]
I guess the relevant code should be in the files src/InstancesEditor/SelectionRectangle.js and src/InstancesEditor/InstancesRotator :)

**Comment by PawBud** [comment]
Thanks @arthuro555 I'll just have a look at it.

**Comment by 4ian** [comment]
Selection rectangles displayed around objects are rendered in https://github.com/4ian/GDevelop/blob/master/newIDE/app/src/InstancesEditor/SelectedInstances.js#L39

Position of the rectangle of an instance is computed using an instance measurer: https://github.com/4ian/GDevelop/blob/c000a735bb45d4f5db8007b27954172e6e5f1095/newIDE/app/src/InstancesEditor/InstancesRenderer/index.js#L33-L75

An `instance` here is a `gd.InitialInstance`, so all its methods here, including the one to get the angle: 
https://github.com/4ian/GDevelop/blob/c000a735bb45d4f5db8007b27954172e6e5f1095/GDevelop.js/Bindings/Bindings.idl#L731-L741

The goal is to render the selection rectangles rotated like the instance is.

**Comment by PawBud** [comment]
If anyone else is interested in issuing a PR; I am not very confident if I'll be able to derive a fix soon so pls go ahead :)

**Comment by VolDemorT-acer** [comment]
@4ian in the previous update ,it was suggested to change rotate selection box as the instance is being rotated but there are two rectangles instancerect and the selection rectangle does the instance rectangle need to be rotated too ??

moreover when we rotate selection rectangle like instance the resize and rotate button need to be rotated as well ??


![Screenshot from 2020-10-09 17-30-22](https://user-images.githubusercontent.com/58597818/95581084-04dd2880-0a56-11eb-98be-2bd5a7c28c6d.png)


Screenshot of rotated selection rectangle and the resize and rotate buttons

**Comment by Bouh** [comment]
Yes all Gizmo should be rotated with it.

Le ven. 9 oct. 2020 à 14:08, VolDemorT-acer <notifications@github.com> a
écrit :

**Comment by VolDemorT-acer** [comment]
#2017

**Comment by 85ayush** [comment]
I would love to work on this issue, I am new to open source contribution, can you please guide me?

**Comment by D8H** [comment]
I think that implementing a shear transformation is a requirement for this because scale * rotation result to an affine transformation with shear.

**Comment by D8H** [comment]
What do you think of something like this?
A selection of several instances is still as useless as before for rotation and scaling, but at least It fells less broken for one instance.
https://user-images.githubusercontent.com/2611977/121887446-05317000-cd17-11eb-8953-0f17afc93052.mp4

**Comment by 85ayush** [comment]
Sorry, but I am busy, these days, so I can not participate in contribution.

**Comment by 4ian** [comment]
> What do you think of something like this?
> A selection of several instances is still as useless as before for rotation and scaling, but at least It fells less broken for one instance.
> https://user-images.githubusercontent.com/2611977/121887446-05317000-cd17-11eb-8953-0f17afc93052.mp4

That's a good improvement I think. Not the perfect way of scaling/rotating multiple objects, but as you said it feels much less "broken" for rotated instances.

**Comment by D8H** [comment]
This version works with several selected instances. The change between the 2 versions is not that big. I think it will be faster to review everything at once.
https://user-images.githubusercontent.com/2611977/122607972-b3bd1400-d07b-11eb-8497-30783dac6b9e.mp4

I will try to add the missing grabbing squares too if it's easy to do.

**Comment by 4ian** [comment]
@D8H Awesome! Let's go for reviewing the whole thing at once :)

**Comment by D8H** [comment]
I'm almost done. I made new videos to show some behavior changes:
* When translating over a grid, the selection will magnet following the cursor direction to be able to magnet to the left or to the right for instance.
It may need a bit of tweaking but it seems usable (I changed the grid from floor to round since the capture, it will impact anything that use the grid, but for the best I hope).
https://user-images.githubusercontent.com/2611977/122686744-1f85b500-d213-11eb-815f-c316d83a477b.mp4

* Same goes for scaling, this video shows the toggling between bottom and right alignment.
https://user-images.githubusercontent.com/2611977/122686747-290f1d00-d213-11eb-8a55-77485159676a.mp4
